3/debian-10/rootfs/opt/bitnami/scripts/libetcd.sh

```diff
@@ -119,6 +119,18 @@ etcdctl_get_endpoints() {
        return 1
     }
 
+    get_ips() {
+        local domain="${1:?domain is required}"
+        local ips=""
+        while is_empty_value "${ips}"; do
+            ips="$(getent ahosts "$domain" | awk '{print $1}' | uniq)"
+            if is_empty_value "${ips}"; then
+              sleep 1
+            fi
+        done
+        echo "${ips}"
+    }
```

Comment:
This approach is very risky since it's likely to enter into an infinite loop. Instead, we can use our `retry_while` function (defined in the [libos.sh library](https://github.com/bitnami/bitnami-docker-etcd/blob/master/3/debian-10/prebuildfs/opt/bitnami/scripts/libos.sh#L269)):
```suggestion
    hostname_has_ips() {
       local hostname="${1:?hostname is required}"
       [[ "$(getent ahosts "$hostname")" != "" ]] && return 0
       return 1
    }
```
Suggestion follow in the next comment

Reply:
I agree. The current implementation is not very safe in that regard. I tested your suggested change and it seems to work 👍🏻